lib/tasks/sanitize_data.rake

```diff
@@ -0,0 +1,162 @@
+      sanitize_batch("StripeAccount", :sanitize_stripe_account)
+
+      # Can't find the model for spree_paypal_accounts table (email and payer_id need to be deleted)
+      #   sanitize_batch("Spree::PaypalAccount", :sanitize_paypal_account, canned_users)
+      ActiveRecord::Base.connection.execute("delete from spree_paypal_accounts")
+
+      # Delete all preferences that may contain sensitive information
+      ActiveRecord::Base.connection.execute("
+        delete
+        from spree_preferences
+        where key like '%gateway%'
+          OR key like '%billing_integration%'
+          OR key like '%s3%'
+        ")
+
+      Spree::LogEntry.destroy_all
+    end
+
+    def sanitize_email(object, sanitized_user)
+      object.email = sanitized_user[:email]
+      object.save!
+    end
+
+    def sanitize_user(user, sanitized_user)
+      user.email = sanitized_user[:email]
+      user.unconfirmed_email = sanitized_user[:email]
+      user.save!
+    end
+
+    def sanitize_customer(customer, sanitized_user)
+      customer.email = sanitized_user[:email]
+      customer.name = sanitized_user[:first_name]
+      customer.save!
+    end
+
+    def sanitize_tokenized_permission(tokenized_permission)
+      tokenized_permission.token = nil
+      tokenized_permission.save!
+    end
+
+    def sanitize_address(address, sanitized_user)
+      return if address.nil? || address.enterprise.present?
+
+      address.firstname = sanitized_user[:first_name]
+      address.lastname = sanitized_user[:last_name]
+      address.phone = sanitized_user[:phone]
+      address.alternative_phone = sanitized_user[:phone]
+      address.address1 = address.address2 = "Dummy address"
+      address.city = "Dummy City"
+      address.zipcode = "0000"
+      address.company = address.latitude = address.longitude = nil
+      address.save!
+    end
+
+    def sanitize_payment_method(payment_method)
+      payment_method.name = "Payment Method #{payment_method.id}"
+      payment_method.description = payment_method.name
+      payment_method.save!
+    end
+
+    def sanitize_credit_card(card)
+      card.month = card.year = card.start_month = card.start_year = 1
+      card.cc_type = card.first_name = card.last_name = card.last_digits = "Dummy"
+      card.save!
+    end
+
+    def sanitize_payment(payment)
+      payment.response_code = payment.avs_response = payment.cvv_response_code = nil
+      payment.identifier = payment.cvv_response_message = nil
+      payment.save!
+    end
+
+    def sanitize_paypal_express_checkout(paypal_express_checkout)
+      paypal_express_checkout.token = nil
+      paypal_express_checkout.save!
+    end
+
+    def sanitize_stripe_account(stripe_account)
+      stripe_account.stripe_user_id = stripe_account.stripe_publishable_key = nil
+      stripe_account.save!
+    end
+
+    def sanitize_batch(model_class, sanitize_method_name, canned_users = nil)
+      model_class.constantize.all.each_with_index do |object, index|
```

Comment:
there is no need to constantize the string if we pass a constant, and AFAIK nothing stops us from doing so.

Reply:
done :+1:
thanks!